src/App.tsx

```diff
@@ -1,35 +1,181 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
-import './App.css'
+import React, { useState, ChangeEvent, useEffect } from 'react';
+import Header from './components/Header';
+import SearchBar from './components/SearchBar';
+import PendingRequests from './components/PendingRequests';
+import MyConnections from './components/MyConnections';
+import { ToastContainer } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import {
+  sendConnectionRequest,
+  getPendingConnectionRequests,
+  findConnectionsByUserId,
+  acceptConnectionRequest,
+  rejectConnectionRequest,
+  removeConnection,
+  searchUser,
+} from './services/api/connections.service';
+import './styles/App.css';
+import './styles/ToastStyles.css';
+import { toast } from 'react-toastify';
 
-function App() {
-  const [count, setCount] = useState(0)
+interface User {
+  id: number;
+  username: string;
+  isSameUser: boolean;
+  hasPendingRequest: boolean;
+  requestId: number;
+  connectionId: number;
+}
```

Comment:
We can create a types file that handles all or specific types, so other components can import the definitions as needed

Reply:
sure also this will help centralize our type definitions and make it easier to manage changes in one place. We can create a types.ts (or similar) file in a shared folder, and components can then import types from there as needed. It’ll also improve code consistency and reduce redundancy across the project.